Content.Server/Paint/PaintSystem.cs

```diff
@@ -0,0 +1,97 @@
+using Content.Shared.Popups;
+using Content.Shared.Paint;
+using Content.Shared.Interaction;
+using Content.Server.Chemistry.Containers.EntitySystems;
+using Robust.Shared.Audio.Systems;
+using Content.Shared.Humanoid;
+using Content.Shared.SubFloor;
+
+namespace Content.Server.Paint;
+
+/// <summary>
+/// Colors target and consumes reagent on each color success.
+/// </summary>
+public sealed class PaintSystem : SharedPaintSystem
+{
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly SolutionContainerSystem _solutionContainer = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<PaintComponent, AfterInteractEvent>(OnInteract);
+    }
+
+    // When paint is used on an entity it will apply the color to the entity provided TryPaint returns true.
+    private void OnInteract(Entity<PaintComponent> entity, ref AfterInteractEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        if (!args.CanReach || args.Target is not { Valid: true } target)
+            return;
+
+        if (!TryComp(args.Used, out PaintComponent? component))
+            return;
+
+        if (TryPaint(entity, target, args.User, args.Used))
+        {
+
+            if (HasComp<AppearanceComponent>(target))
+                RemComp<AppearanceComponent>(target);
+
+            EnsureComp<PaintedComponent>(target);
+            AddComp<AppearanceComponent>(target);
+
+            if (!TryComp(target, out PaintedComponent? paint))
+                return;
+            else
```

Comment:
EnsureComp returns the component TryComp is unnecessary.

Reply:
trycomp is removed.